RPA.MIT.Notification.Function/Notification.cs

```diff
@@ -3,143 +3,146 @@
             _eventQueueService = eventQueueService;
+            _logger = loggerFactory.CreateLogger<Notification>();
         }
 
-        [FunctionName("SendNotification")]
+        [Function("SendNotification")]
         public async Task CreateEvent(
-            [QueueTrigger("invoicenotification", Connection = "QueueConnectionString")] string notificationMsg,
-            [Table("invoicenotification", Connection = "TableConnectionString")] TableClient tableClient,
-            ILogger log)
+            [QueueTrigger("invoicenotification", Connection = "QueueConnectionString")] string notificationMsg)
         {
-            log.LogInformation("MIT Notification queue trigger function processing: {notificationMsg}", notificationMsg);
+            _logger.LogInformation("MIT Notification queue trigger function processing: {notificationMsg}", notificationMsg);
 
             try
             {
                 var isValid = ValidateMessage.IsValid(notificationMsg);
 
                 if (!isValid)
                 {
-                    log.LogError("Invalid message: {notificationMsg}", notificationMsg);
-                    return;
+                    _logger.LogError("Invalid message: {notificationMsg}", notificationMsg);
                 }
 
                 dynamic notificationMsgObj = JObject.Parse(notificationMsg);
                 string templateName = notificationMsgObj.Action;
                 string scheme = notificationMsgObj.Scheme;
-                var templateId = _configuration[$"templates:{templateName}"];
-                var emailAddress = _configuration[$"schemas:{scheme}"];
+                var templateId = _configuration[$"templates{templateName}"];
+                var emailAddress = _configuration[$"schemas{scheme}"];
                 string id = notificationMsgObj.Id;
 
                 if (templateId == null)
                 {
-                    log.LogError("Template not found for action: {action}", templateName);
+                    _logger.LogError("Template not found for action: {action}", templateName);
                     await _eventQueueService.CreateMessage(id, "failed", "notification", "Template not found", notificationMsg);
-                    return;
                 }
 
                 if (emailAddress == null)
                 {
-                    log.LogError("emailAddress not found for scheme: {scheme}", scheme);
+                    _logger.LogError("emailAddress not found for scheme: {scheme}", scheme);
                     await _eventQueueService.CreateMessage(id, "failed", "notification", "emailAddress not found", notificationMsg);
-                    return;
                 }
 
-                log.LogInformation("Sending email for incoming message id: {id}", id);
+                _logger.LogInformation("Sending email for incoming message id: {id}", id);
 
                 var notifyResponse = _notifyService.SendEmail(emailAddress, templateId, notificationMsgObj.Data);
```

Comment:
I wrote a very similar service to this last year, I used Polly (https://github.com/App-vNext/Polly) to handle retries and timeouts on the external calls.

Reply:
Code updated to use correct logging.